operator/controller/src/main/java/io/apicurio/registry/operator/feat/TLS.java

```diff
@@ -0,0 +1,67 @@
+package io.apicurio.registry.operator.feat;
+
+import io.apicurio.registry.operator.api.v1.ApicurioRegistry3;
+import io.apicurio.registry.operator.api.v1.ApicurioRegistry3Spec;
+import io.apicurio.registry.operator.api.v1.spec.AppSpec;
+import io.apicurio.registry.operator.api.v1.spec.TLSSpec;
+import io.apicurio.registry.operator.utils.SecretKeyRefTool;
+import io.fabric8.kubernetes.api.model.EnvVar;
+import io.fabric8.kubernetes.api.model.apps.Deployment;
+
+import java.util.Map;
+import java.util.Optional;
+
+import static io.apicurio.registry.operator.EnvironmentVariables.*;
+import static io.apicurio.registry.operator.resource.app.AppDeploymentResource.addEnvVar;
+import static java.util.Optional.ofNullable;
+
+public class TLS {
+
+    public static void configureTLS(ApicurioRegistry3 primary, Deployment deployment,
+                               String containerName, Map<String, EnvVar> env) {
+
+        addEnvVar(env, QUARKUS_HTTP_INSECURE_REQUESTS, Optional.ofNullable(primary.getSpec())
+                .map(ApicurioRegistry3Spec::getApp)
+                .map(AppSpec::getTls)
+                .map(TLSSpec::getInsecureRequests)
+                .orElse("enabled"));
+
+        var keystore = new SecretKeyRefTool(getTlsSpec(primary)
+                .map(TLSSpec::getKeystoreSecretRef)
+                .orElse(null), "user.p12");
+
+        var keystorePassword = new SecretKeyRefTool(getTlsSpec(primary)
+                .map(TLSSpec::getKeystorePasswordSecretRef)
+                .orElse(null), "user.password");
+
+        var truststore = new SecretKeyRefTool(getTlsSpec(primary)
+                .map(TLSSpec::getTruststoreSecretRef)
+                .orElse(null), "ca.p12");
+
+        var truststorePassword = new SecretKeyRefTool(getTlsSpec(primary)
+                .map(TLSSpec::getTruststorePasswordSecretRef)
+                .orElse(null), "ca.password");
+
+        if (truststore.isValid() && truststorePassword.isValid()) {
+            // ===== Truststore
+            truststore.applySecretVolume(deployment, containerName);
+            addEnvVar(env, QUARKUS_TLS_TRUST_STORE_P12_PATH, truststore.getSecretVolumeKeyPath());
+            truststorePassword.applySecretEnvVar(env, QUARKUS_TLS_TRUST_STORE_P12_PASSWORD);
+        }
```

Comment:
what if I will use just PEM cert as truststore? Like
```
kind: ApicurioRegistry3
apiVersion: registry.apicur.io/v1
metadata:
  name: schema-registry
spec:
  app:
    storage:
      type: "kafkasql"
      kafkasql:
        bootstrapServers: anubis-kafka-bootstrap.strimzi-kafka.svc:9093
        tls:
          truststoreSecretRef:
            name: anubis-cluster-ca-cert
            key: ca.crt
```
wouldn't it cause failure? I would more expect that keystore password will be required

Reply:
At the operator level we're supporting P12 (since it's usually a better practice than just specifying cert by cert). You will still be able to use PEM files using Quarkus own env vars.

Reply:
I see, thanks.